pagefaultgames/pokerogue
# [Localization] Localize hp phases and translate in Korean

## What are the changes?
Add the option to localize message text in phases.ts, 2 HP restore messages.

## Why am I doing these changes?
All text should be localizable and I thought HP message need to translate.

## What did change?
 - Change the code for localize -> messages of phases.ts to add the option to localize
 - Added localization key in each language folders.
 - Translated part of them to Korean.
 - Note : Localized messages are below
```
  "turnEndHpRestore": "{{pokemonName}}'s HP was restored.", // when pokemon healed by TerrainType.GRASSY (GRASSY_TERRAIN)
  "hpIsFull": "{{pokemonName}}'s\nHP is full!", // when pokemon's HP already full
```

### Screenshots/Videos

English message should not be changed.
(If images are too small, just click image)

| Case | Language | As-is | To-be |
| :-----: | :-----: | :-----: | :---: |
| turnEndHpRestore | En | ![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/857a1e74-69d1-42b8-b7f7-9210fede7e3f) | ![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/e4be8d96-50c6-4975-ad17-78693116c735) |
| turnEndHpRestore | Ko | ![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/b1e09042-b0e4-42c5-ac09-6ab46dbf5777) | ![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/8d2cefe9-4b9f-44fe-a230-b710e2550ed7) |
| turnEndHpRestore | Fr | ![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/124c66ef-64ce-4fd5-bf43-299801fd21a5) | ![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/f239c628-4a18-4fc8-947d-d7a915ab16b3) |
| hpIsFull | En | ![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/bb224f01-06f8-4880-a92b-3437655cac8a) | ![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/7219d8e4-1532-4d9e-b290-c38afc612bd0) |
| hpIsFull | Ko | ![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/9d525d8f-6886-463f-91ea-e06d88c43ebc) | ![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/e0441f49-2684-43e8-919a-240ed871f78e) |
| hpIsFull | Fr | ![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/9e3921c9-e8a1-44f9-a80f-8d2db838c2d4) | ![image](https://github.com/pagefaultgames/pokerogue/assets/19601207/8c1dcf8e-f55a-4c74-9304-42f956a615a0) |

## How to test the changes?
Manually.
Change override.ts moveset, add Grassy Terrain to make terrain Grassy.
```typescript
export const MOVESET_OVERRIDE: Array<Moves> = [ Moves.GRASSY_TERRAIN ];
```
Then check when turn end.
hpIsFull message is occur when pokemon's HP is full.

## Checklist
